core/src/main/scala/org/apache/spark/Accumulable.scala

```diff
@@ -56,15 +61,26 @@ class Accumulable[R, T] private (
     @transient private val initialValue: R,
     param: AccumulableParam[R, T],
     val name: Option[String],
-    private[spark] val countFailedValues: Boolean)
+    private[spark] val countFailedValues: Boolean,
+    private[spark] val dataProperty: Boolean)
   extends Serializable {
 
   private[spark] def this(
       initialValue: R,
       param: AccumulableParam[R, T],
       name: Option[String],
+      countFailedValues: Boolean,
+      dataProperty: Boolean) = {
+    this(AccumulatorContext.newId(), initialValue, param, name, countFailedValues, dataProperty)
+  }
+
+  private[spark] def this(
+      initialValue: R,
+      param: AccumulableParam[R, T],
+      name: Option[String],
       countFailedValues: Boolean) = {
-    this(AccumulatorContext.newId(), initialValue, param, name, countFailedValues)
+    this(AccumulatorContext.newId(), initialValue, param, name, countFailedValues,
+      false /* dataProperty */)
```

Comment:
So for reasons (which aren't super clear), we can't specify it this way (you can see Josh ran into similar troubles in some other places) - but since we aren't going to deal with the old API I'll just rip this part out anyways.